src/Slices/Settings/UI/Tabs/Configuration/Tab.test.tsx

```diff
@@ -281,6 +281,400 @@ describe("ConfigurationTab", () => {
+        protected: true,
+        protected_by: "project.yml",
+      },
+    };
+
+    server.use(
+      http.get("/api/v2/environment_settings", () => {
+        return HttpResponse.json({
+          data: {
+            settings: EnvironmentSettings.base.settings,
+            definition: overridenDefinition,
+          },
+        });
+      })
+    );
+
+    const { component } = setup();
+
+    render(component);
+
+    const row = await screen.findByRole("row", {
+      name: "Row-autostart_agent_map",
+    });
+
+    // Wait for the protected message to appear, indicating the component is properly loaded
+    await screen.findByText("This setting is protected by project.yml, and cannot be changed.");
+
+    const newEntryRow = within(row).getByRole("row", { name: "Row-newEntry" });
+    const newKeyInput = within(newEntryRow).getByRole("textbox", {
+      name: "editEntryKey",
+    });
+    const newValueInput = within(newEntryRow).getByRole("textbox", {
+      name: /editentryvalue/i,
+    });
+
+    expect(newKeyInput).toBeDisabled();
+    expect(newValueInput).toBeDisabled();
+
+    await act(async () => {
+      const results = await axe(document.body);
+
+      expect(results).toHaveNoViolations();
+    });
+  });
+
+  test("GIVEN ConfigurationTab WITH protected enum field THEN disables editing", async () => {
+    const overridenDefinition = {
+      ...EnvironmentSettings.base.definition,
+      agent_trigger_method_on_auto_deploy: {
+        ...EnvironmentSettings.base.definition.agent_trigger_method_on_auto_deploy,
+        protected: true,
+        protected_by: "project.yml",
+      },
+    };
+    server.use(
+      http.get("/api/v2/environment_settings", () => {
+        return HttpResponse.json({
+          data: {
+            settings: EnvironmentSettings.base.settings,
+            definition: overridenDefinition,
+          },
+        });
+      })
+    );
+
+    const { component } = setup();
+
+    render(component);
+
+    const row = await screen.findByRole("row", {
+      name: "Row-agent_trigger_method_on_auto_deploy",
+    });
+
+    await userEvent.click(
+      within(row).getByRole("combobox", {
+        name: "EnumInput-agent_trigger_method_on_auto_deployFilterInput",
+      })
+    );
+
+    // Check that the toggle button (which contains the combobox) is disabled by CSS class
+    const toggle = within(row).getByTestId("EnumInput-agent_trigger_method_on_auto_deploy-toggle");
+    expect(toggle).toHaveClass("pf-m-disabled");
+
+    await act(async () => {
+      const results = await axe(document.body);
+
+      expect(results).toHaveNoViolations();
+    });
+  });
+
+  test("GIVEN ConfigurationTab WITH protected boolean field THEN disables editing", async () => {
+    const overridenDefinition = {
+      ...EnvironmentSettings.base.definition,
+      auto_deploy: {
+        ...EnvironmentSettings.base.definition.auto_deploy,
+        protected: true,
+        protected_by: "config.yml",
```

Comment:
I'm not sure if the `protected_by` field can ever be something other than `project.yml` or `Null` but this is a minor detail.